idseq_dag/steps/run_srst2.py

```diff
@@ -69,6 +72,76 @@ def get_common_params(self):
         return ['--min_coverage', min_cov,'--threads', n_threads,
                 '--output',  os.path.join(self.output_dir_local, 'output'), '--log', '--gene_db', db_file_path]
 
+    def generate_mapped_reads_table(self):
+        """Use bedtools to generate a table of mapped reads for each genome in the ARG ANNOT database.
+            If a new resistance gene db is used, the .bed file will need to be updated manually."""
+        bed_file_path = fetch_from_s3(self.additional_files["resist_genome_bed"], self.output_dir_local, allow_s3mi=False)
+        bedtools_params = ['bedtools', 'coverage', '-b', self.output_files_local()[5], '-a', bed_file_path, '>>', os.path.join(self.output_dir_local, 'matched_reads.tsv')]
+        command.execute(" ".join(bedtools_params))
+
+    def get_total_reads(self, isZipped, isFasta):
+        """Gets the total number of reads in the sample by counting them directly from the
+            fastq or fasta files."""
+        input_filenames = self.input_files_local[0]
+        if isZipped:
```

Comment:
Can you add a TODO to factor this out into a utility function? nonhost_fastq has a similar function.